### PR TITLE
feat : Core contracts with foundry

### LINKED ISSUE #1448: Core contracts in foundry
**Description**
Write deployment scripts and tests with Foundry for core contracts. 

**Motivation**
Bring governance contracts into monorepo. 

**Implementation details** 

- Figure out if deployment with proxies is possible. 
- Test contracts with Foundry.

### PR DESCRIPTION
**Pull request by mrhouzlane**
## Description

Integrate Core contracts and Governance contracts using Foundry instead of Hardhat.   

#### Add scripts : 
- Deploy Core 
- Deploy Proxies 
- Upgrade Proxies 

#### Add tests for : 
- **EscrowFactory**
- **Escrow**
- **Staking** 
- **KVStore**
- **RewardPool** 

#### Add workflow for contract deployment 

Closes #1448

### CODE REVIEW

File: package-lock.json
Comment (portuu3) [inline package-lock.json:1]: Remove this file

File: packages/core/package.json [hunk 1]
```diff
@@ -1,93 +1,26 @@
...
-    "./abis/*": "./abis/*",
-    "./artifacts/*": "./artifacts/*",
-    "./typechain-types": {
-      "default": "./dist/typechain-types/index.js",
-      "require": "./dist/typechain-types/index.js",
-      "import": "./typechain-types/index.ts",
-      "types": "./dist/typechain-types/index.d.ts"
+    "name": "core-human-contracts",
+    "version": "1.0.0",
+    "description": "Core Human Protocol contracts ",
```
Comment (portuu3) [inline packages/core/package.json:4]:
```suggestion
    "description": "Human Protocol Core Smart Contracts",
```

File: packages/core/package.json [hunk 2]
```diff
@@ -1,93 +1,26 @@
...
-  },
-  "exports": {
-    "./abis/*": "./abis/*",
-    "./artifacts/*": "./artifacts/*",
-    "./typechain-types": {
-      "default": "./dist/typechain-types/index.js",
-      "require": "./dist/typechain-types/index.js",
-      "import": "./typechain-types/index.ts",
-      "types": "./dist/typechain-types/index.d.ts"
+    "name": "core-human-contracts",
```
Comment (portuu3) [inline packages/core/package.json:2]:
```suggestion
    "name": "@human-protocol/core",
```

File: packages/core/foundry.toml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+[profile.default]
+src = "src"
+out = "out"
+libs = ["lib"]
+via-ir=true
+optimizer_runs = 10
+gas_reports = ["*"]
+
+
+[rpc_endpoints]
```
Comment (portuu3) [inline packages/core/foundry.toml:10]: remove unnecessary networks

File: packages/core/foundry.toml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+gnosisChiado = "https://rpc.chiadochain.net"
+okcTest = "https://exchaintestrpc.okex.org"
+
+[etherscan]
+polygonMumbai = { key = "${ETHERSCAN_API_KEY}" }
+goerli = { key = "${ETHERSCAN_API_KEY}" }
+arbitrumGoerli = { key = "${ETHERSCAN_API_KEY}" }
+baseGoerli = { key = "${ETHERSCAN_API_KEY}" }
+celoAlfajores = { key = "${ETHERSCAN_API_KEY}" }
+scrollSepolia = { key = "${ETHERSCAN_API_KEY}" , url = "https://api-sepolia.scrollscan.dev/api" }
```
Comment (portuu3) [inline packages/core/foundry.toml:40]: remove

File: packages/core/test.md
Comment (portuu3) [inline packages/core/test.md:1]: Remove this file

File: packages/core/sample-spoke-params.json
Comment (portuu3) [inline packages/core/sample-spoke-params.json:1]: what is this?

File: packages/core/src/hm-token/HMToken.sol
Comment (portuu3) [inline packages/core/src/hm-token/HMToken.sol:1]: we have two versions of HM Token

**Review by portuu3 — CHANGES_REQUESTED** [review]

File: packages/core/sample-spoke-params.json
Comment (mrhouzlane) [reply packages/core/sample-spoke-params.json:1]: this is from governance

File: packages/core/src/hm-token/HMToken.sol
Comment (mrhouzlane) [reply packages/core/src/hm-token/HMToken.sol:1]: removed duplicate

File: packages/core/package.json (hunk 2 above)
Comment (mrhouzlane) [reply packages/core/package.json]: change done

File: packages/core/package.json (hunk 1 above)
Comment (mrhouzlane) [reply packages/core/package.json]: change done

**Pull request closed without merging**